src/ChemicalDiffusion/ChemicalDiffusion.jl

```diff
@@ -0,0 +1,349 @@
+    ΔV_2σ_SI       = uconvert(m^3 / mol, pp.ΔV_2σ)
+    T_range_min_SI = uconvert(K, T_range_min)
+    T_range_max_SI = uconvert(K, T_range_max)
+    P0_SI          = uconvert(Pa, P0)
+
+
+    args       = (Name=unsafe_string(pp.Name),
+                  Mineral=unsafe_string(pp.Mineral),
+                  Formula=unsafe_string(pp.Formula),
+                  Species=unsafe_string(pp.Species),
+                  D0=D0_SI, D0_2σ=D0_2σ_SI,
+                  Ea=Ea_SI, Ea_2σ=Ea_2σ_SI,
+                  ΔV=ΔV_SI, ΔV_2σ=ΔV_2σ_SI,
+                  T_range_min_SI=T_range_min_SI,
+                  T_range_max_SI=T_range_max_SI,
+                  P0=P0_SI,
+                  Orientation=unsafe_string(pp.Orientation),
+                  Crystallography=unsafe_string(pp.Crystallography),
+                  Buffer=unsafe_string(pp.Buffer),
+                  Fluid=unsafe_string(pp.Fluid)
+                  )
+
+    return DiffusionData(; args...)
+end
+
+function Transform_ChemicalDiffusion(pp::AbstractChemicalDiffusion{T}, kwargs::NamedTuple) where T
+
+    f(a,b) = Value(GeoUnit(a))
+    f(::Nothing,b) = Value(b)
+
+    (; D0, D0_2σ, Ea, Ea_2σ, ΔV, ΔV_2σ, T_range_min, T_range_max, P0) = kwargs
+
+    D0_new          = f(D0, pp.D0)
+    D0_2σ_new       = f(D0_2σ, pp.D0_2σ)
+    Ea_new          = f(Ea, pp.Ea)
+    Ea_2σ_new       = f(Ea_2σ, pp.Ea_2σ)
+    ΔV_new          = f(ΔV, pp.ΔV)
+    ΔV_2σ_new       = f(ΔV_2σ, pp.ΔV_2σ)
+    T_range_min_new = f(T_range_min, pp.T_range_min)
+    T_range_max_new = f(T_range_max, pp.T_range_max)
+    P0_new          = f(P0, pp.P0)
+
+    D0_SI          = uconvert(m^2 / s, D0_new)
+    D0_2σ_SI       = uconvert(m^2 / s, D0_2σ_new)
+    Ea_SI          = uconvert(J / mol, Ea_new)
+    Ea_2σ_SI       = uconvert(J / mol, Ea_2σ_new)
+    ΔV_SI          = uconvert(m^3 / mol, ΔV_new)
+    ΔV_2σ_SI       = uconvert(m^3 / mol, ΔV_2σ_new)
+    T_range_min_SI = uconvert(K, T_range_min_new)
+    T_range_max_SI = uconvert(K, T_range_max_new)
+    P0_SI          = uconvert(Pa, P0_new)
+
+    args       = (Name=unsafe_string(pp.Name),
+                  Mineral=unsafe_string(pp.Mineral),
+                  Formula=unsafe_string(pp.Formula),
+                  Species=unsafe_string(pp.Species),
+                  D0=D0_SI, D0_2σ=D0_2σ_SI,
+                  Ea=Ea_SI, Ea_2σ=Ea_2σ_SI,
+                  ΔV=ΔV_SI, ΔV_2σ=ΔV_2σ_SI,
+                  T_range_min=T_range_min_SI,
+                  T_range_max=T_range_max_SI,
+                  P0=P0_SI,
+                  Orientation=unsafe_string(pp.Orientation),
+                  Crystallography=unsafe_string(pp.Crystallography),
+                  Buffer=unsafe_string(pp.Buffer),
+                  Fluid=unsafe_string(pp.Fluid)
+                  )
+
+    return DiffusionData(; args...)
+end
+
+export SetChemicalDiffusion,
+       Transform_ChemicalDiffusion
+
+
+# load collection of chemical diffusion data
+include("Data/Rutile/Rutile.jl")
+using .Rutile
+
+export Rutile
```

Comment:
it's better to put all the `include`, `using` and `export` at the top of the module